source/jormungandr/jormungandr/interfaces/v1/Coord.py

```diff
@@ -28,60 +28,72 @@
 from collections import OrderedDict
 import datetime
+from jormungandr.utils import is_coord, get_lon_lat
+import six
 
 
 class Coord(ResourceUri):
+    def __init__(self, *args, **kwargs):
+        ResourceUri.__init__(self, *args, **kwargs)
+        self.parsers['get'].add_argument("_autocomplete", type=six.text_type, hidden=True,
+                                         help="name of the autocomplete service, used under the hood")
 
-    def get(self, region=None, lon=None, lat=None, id=None, *args, **kwargs):
-        if id is not None:
-            splitted = id.split(";")
-            if len(splitted) != 2:
-                abort(404, message='invalid coords [{}], should be <lon:lon>;<lat:lat>'.format(id))
-            lon, lat = splitted
-            try:
-                lon = float(lon)
-                lat = float(lat)
-            except ValueError:
-                abort(404, message='invalid coords [{}], should be <lon:lon>;<lat:lat>'.format(id))
-
-        if region is None:
-            regions = i_manager.get_regions("", lon, lat)
-        else:
-            regions = [region]
+    def _get_regions(self, region=None, lon=None, lat=None):
+        return [region] if region else i_manager.get_regions("", lon, lat)
 
+    def _get_args(self, lon=None, lat=None, id=None):
         args = {
-            "uri": "coord:{}:{}".format(lon, lat),
+            "uri": "{};{}".format(lon, lat),
+            "_current_datetime": datetime.datetime.utcnow()
+        }
+        if all([lon, lat, is_coord(id)]):
+            return args
+        args.update({
             "count": 1,
             "distance": 200,
             "type[]": ["address"],
             "depth": 1,
             "start_page": 0,
-            "filter": "",
-            "_current_datetime": datetime.datetime.utcnow()
-        }
+            "filter": ""
+        })
+        return args
+
+    def get(self, region=None, lon=None, lat=None, id=None):
+        args = self.parsers["get"].parse_args()
+
+        if is_coord(id):
+            lon, lat = get_lon_lat(id)
+
+        regions = self._get_regions(region=region, lon=lon, lat=lat)
 
-        self._register_interpreted_parameters(args)
         result = OrderedDict()
 
+        params = self._get_args(id=id, lon=lon, lat=lat)
+        args.update(params)
+        self._register_interpreted_parameters(args)
         for r in regions:
             self.region = r
             result.update(regions=[r])
-            pb_result = i_manager.dispatch(args, "places_nearby", instance_name=r)
-            if len(pb_result.places_nearby) > 0:
-                e_type = pb_result.places_nearby[0].embedded_type
-                if _NAVITIATYPE.values_by_name["ADDRESS"].number == e_type:
-                    new_address = marshal(pb_result.places_nearby[0].address,
-                                          address)
-                    result.update(address=new_address)
+            if all([lon, lat, is_coord(id)]):
```

Comment:
Are you sure it's restrictive enough? I wonder about an url being `v1/coords/{lon;lat}/departures`.
Maybe this test already exists though (or would be nice to have it).

Reply:
"places_nearby" is called in this case